File: CS/Task.cs
using System;
using System.ComponentModel;

namespace DXSample {
    public class Task : IDataErrorInfo {
        public string TaskName { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }

        string IDataErrorInfo.Error => StartDate > EndDate ? "Either Start Date or End Date should be corrected." : null;
        string IDataErrorInfo.this[string columnName] {
            get {
                switch (columnName) {
                    case nameof(StartDate):
                        if (StartDate > EndDate)
                            return "Start Date must be less than End Date";
                        break;
                    case nameof(EndDate):
                        if (StartDate > EndDate)
                            return "End Date must be greater than Start Date";
                        break;
                    case nameof(TaskName):
                        if (string.IsNullOrEmpty(TaskName))
                            return "Enter a task name";
                        break;
                }
                return null;
            }
        }
    }
}

File: VB/Task.vb
Imports System.ComponentModel

Namespace DXSample

    Public Class Task
        Implements IDataErrorInfo

        Public Property TaskName As String

        Public Property StartDate As Date

        Public Property EndDate As Date

        Private ReadOnly Property [Error] As String Implements IDataErrorInfo.[Error]
            Get
                Return If(StartDate > EndDate, "Either Start Date or End Date should be corrected.", Nothing)
            End Get
        End Property

        Private ReadOnly Property Item(ByVal columnName As String) As String Implements IDataErrorInfo.Item
            Get
                Select Case columnName
                    Case NameOf(Task.StartDate)
                        If StartDate > EndDate Then Return "Start Date must be less than End Date"
                    Case NameOf(Task.EndDate)
                        If StartDate > EndDate Then Return "End Date must be greater than Start Date"
                    Case NameOf(Task.TaskName)
                        If String.IsNullOrEmpty(TaskName) Then Return "Enter a task name"
                End Select

                Return Nothing
            End Get
        End Property
    End Class
End Namespace
